### PR TITLE
[CODEGEN] fix vector conversion for opencl

### PR DESCRIPTION
**Pull request by merrymercy**
1. fix vector conversion for opencl
2. add more argument type in depthwise_conv2d
3. when mark pointer type as 'restrict', use 'float32' as default type. I am not sure whether I do it correctly.

### CODE REVIEW

File: src/pass/split_host_device.cc [hunk 1]
```diff
@@ -192,8 +192,8 @@ class HostDeviceSplitter : public IRMutator {
         if (it != handle_data_type_.end()) {
           n->handle_data_type.Set(v, it->second);
         } else {
-          // int32 as a placeholder
-          n->handle_data_type.Set(v, make_const(UInt(32), 0));
+          // mark all the pointer argument 'restrict', use float32 as default type
```
Comment (tqchen) [inline src/pass/split_host_device.cc]: Can you give a case when we hit this codepath?

File: topi/python/topi/nn/depthwise_conv2d.py [hunk 2]
```diff
@@ -131,7 +139,10 @@ def depthwise_conv2d_backward_input_nhwc(Filter, Out_grad, oshape, ishape, strid
     batch, in_h, in_w, in_c = ishape
     _, out_h, out_w, out_c = oshape
     filter_h, filter_w, _, channel_multiplier = Filter.shape
-    stride_h, stride_w = stride
+    if isinstance(stride, int):
```
Comment (tqchen) [inline topi/python/topi/nn/depthwise_conv2d.py]: add a unit test-case that cover this code path

Comment (merrymercy) [reply topi/python/topi/nn/depthwise_conv2d.py]: modified the test script to use both types of argument.

File: src/pass/split_host_device.cc (hunk 1 above)
Comment (merrymercy) [reply src/pass/split_host_device.cc]: remove this and update the codegen

**Comment by merrymercy** [comment]
@tqchen addressed the comment

**Pull request merged**